tests/manage/z_cluster/cluster_expansion/test_delete_pod.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+import logging
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework.testlib import ignore_leftovers, tier4a
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources import pod as pod_helpers
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.ocs.cluster import get_percent_used_capacity
+from ocs_ci.utility.utils import ceph_health_check
+from tests.disruption_helpers import Disruptions
+
+
+@ignore_leftovers
+@tier4a
+class TestAddCapacityWithResourceDelete:
+    """
+    Test add capacity when one of the resources gets deleted
+    in the middle of the process.
+    """
+
+    new_pods_in_status_running = False
+
+    def kill_resource_repeatedly(self, resource_name, resource_id, max_iterations=30):
+        """
+        The function get the resource name, and id and kill the resource repeatedly
+        until the new osd pods reached status running.
+        Args:
+            resource_name (str): the name of the resource to kill
+            resource_id (int): the id of the resource to kill
+            max_iterations (int): Maximum times of iterations to delete the given resource
+        """
+        d = Disruptions()
+        logging.info("starting function 'kill_resource_repeatedly'")
```

Comment:
im not sure this log needed in production

Reply:
fixed